src/server.py

```diff
@@ -27,38 +35,130 @@ def server():
             if len(part) < buffer_length:
                 message_complete = True
         print(message)
-        conn.send(response_ok())
+
+        try:
+            uri = parse_request(message)
+        except HTTPException as exception:
+            msg = response_error(exception)
+        try:
+            file_data_tuple = resolve_uri(uri)
+        except HTTPException as exception:
+            msg = response_error(exception)
+        try:
+            msg = response_ok(file_data_tuple)
+        except HTTPException:
+            msg = (response_error("500 Internal Server Error"))
+
+        conn.sendall(msg.encode("utf-8"))
         conn.close()
 
 
-def response_ok():
-    """Function returns an HTTP '200 OK' response."""
-    response = (b"HTTP/1.1 200 OK\r\n"
-                b"Host: 127.0.0.1:5000\r\n\r\n"
-                b"Connection Successful")
-    return response
+def resolve_uri(uri):
+    if uri.find('../') is not -1:
+        raise HTTPException('403 Forbidden')
+    elif uri[-1] == '/':
```

Comment:
Are you certain that requests for a folder will always end in a slash?  If not, is this a safe validation?